src/runtime/pickle_runner.ts

```diff
@@ -375,9 +411,33 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+
+    let stepResult
+    const beforeStepHookResults = await this.runStepHooks(
+      this.getBeforeStepHookDefinitions(),
+      stepResult
     )
+    stepResult = new Query().getWorstTestStepResult(beforeStepHookResults)
+    if (stepResult.status !== Status.FAILED) {
+      stepResult = await this.invokeStep(
+        testStep.pickleStep,
+        testStep.stepDefinitions[0]
+      )
+    }
+    const afterStepHookResults = await this.runStepHooks(
+      this.getAfterStepHookDefinitions(),
+      stepResult
+    )
+    const stepHookResults = beforeStepHookResults.concat(afterStepHookResults)
+    stepResult.status = new Query().getWorstTestStepResult(
+      stepHookResults.concat(stepResult)
+    ).status
```

Comment:
I think we need to assign more than just status here (things like message are also important). And I think the duration below should be recalculated from scratch (if a before hook fails I believe its duration would be counted twice)

Thoughts on 
```ts
const stepResults = /* run before step hook */;
// if worst step test result status is not failed, run step and append to stepResults
// concat after step hook results to step results
const finalStepResult = /* worst step test result */;
finalStepResult.duration = getZeroDuration();
// iterate though step results, incrementing duration
return finalStepResult;
```

Reply:
Updated to address this. There's two weird things I found while doing this:

1) If I concatenate the afterStepHook Results into the stepResults, then if an after step hook failed it wasn't getting picked up. I could only get it to pickup if I manually in a for loop pushed each individual result into the stepResults array. Not sure why this is, but this was the only way I could get it to work.

2) Had to make a separate variable to sum durations into and then assign that to the finalStepResult. If I just did what you suggested, I kept running into either getting twice the duration, or always getting zero duration. I tried a lot of different ways of doing it, but what I have is the only one I could actually get to work properly.

Reply:
@davidjgoss  Would you also have any idea as to my questions on this thread?

Reply:
Sorry for late reply - I should be able to have a look at this today/tonight.

Reply:
For concat, did you reassign the variable? (concat returns the new array instead of altering the variable)

For the sum durations, current implementation looks fine.

Reply:
Regarding the durations, can you please add a unit test for that?

Reply:
Yup, forgot to reassign the variable for concat, not sure how I missed that.

Also added a unit test with the other duration ones in summary_helper_spec.ts

Reply:
@charlierudolph Are we finally good to merge this PR?

Reply:
Yep. Looks good! Thanks for your contribution!